src/main/java/uk/nhs/tis/trainee/usermanagement/config/SecurityConfig.java

```diff
@@ -0,0 +1,102 @@
+import com.transformuk.hee.tis.security.JwtAuthenticationSuccessHandler;
+import com.transformuk.hee.tis.security.RestAccessDeniedHandler;
+import com.transformuk.hee.tis.security.filter.JwtAuthenticationTokenFilter;
+import java.util.Collections;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.ProviderManager;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@EnableAutoConfiguration
+@Import(JwtSpringSecurityConfig.class)
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class SecurityConfig {
+
+  private static final String API_PATH = "/api/**";
+
+  private final JwtAuthenticationEntryPoint unauthorizedHandler;
+  private final RestAccessDeniedHandler accessDeniedHandler;
+  private final JwtAuthenticationProvider authenticationProvider;
+
+  SecurityConfig(JwtAuthenticationEntryPoint unauthorizedHandler,
+      RestAccessDeniedHandler accessDeniedHandler,
+      JwtAuthenticationProvider authenticationProvider) {
+    this.unauthorizedHandler = unauthorizedHandler;
+    this.accessDeniedHandler = accessDeniedHandler;
+    this.authenticationProvider = authenticationProvider;
+  }
+
+  @Bean
+  public AuthenticationManager authenticationManager() {
+    return new ProviderManager(Collections.singletonList(authenticationProvider));
+  }
+
+  @Bean
+  public JwtAuthenticationTokenFilter authenticationTokenFilterBean() {
+    JwtAuthenticationTokenFilter authenticationTokenFilter = new JwtAuthenticationTokenFilter(
+        API_PATH);
+    authenticationTokenFilter.setAuthenticationManager(authenticationManager());
+    authenticationTokenFilter
+        .setAuthenticationSuccessHandler(new JwtAuthenticationSuccessHandler());
+    return authenticationTokenFilter;
+  }
+
+  @Bean
+  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+    http
+        // we don't need CSRF because our token is invulnerable
+        .csrf().disable()
+        .authorizeRequests()
+        .antMatchers(HttpMethod.GET, API_PATH).hasAuthority("trainee-support:view")
+        .antMatchers(HttpMethod.POST, API_PATH).hasAuthority("trainee-support:modify")
```

Comment:
@DorisWongNHS This is the part that will block access to your new endpoint without the `trainee-support:modify` permission being added to tis-profile